retro/retro.asm

```diff
@@ -476,28 +476,35 @@ endif
 ; If there is an attempt to select a non-existent drive, SELDSK returns
 ; HL=0000H as an error indicator.
 ;
-; The Z80 Retro! only has one drive.
-;
-;##########################################################################
-.bios_seldsk:
-	ld	a,c
-	ld	(bios_disk_disk),a		; XXX should we save this now or defer 
-					; till after we validate the value?
-
+; The Z80 Retro! only has 4 disks
+;
+; Modified for 4 disks (A,B,C & D) by Trevor Jacobs 02/16/2023
+;##########################################################################
+.bios_seldsk:		;select disk given by bc - from alteration guide p26
+	ld	a,c 
+	ld	hl,0		; HL = 0 = invalid disk
+	cp	4			; highest disk number+1 (valid disk # = 0-3)
+	ret nc			; if disk number is not valid return error
+	;
+	ld	(bios_disk_current_disk),a	;safe to update disk number - no error
+	;
 if .debug >= 2
 	call	iputs
 	db	".bios_seldsk entered: \0"
 	call	bios_debug_disk
 endif
-
-	ld	hl,0			; HL = 0 = invalid disk 
-	ld	a,(bios_disk_disk)
-	or	a			; did drive A get selected?
-	ret	nz			; no -> error
-
-	ld	hl,.bios_dph		; the DPH for disk A
+	;
+	ld	a,(bios_disk_current_disk)	;a gets destoyed by debug so must reload
```

Comment:
Move this line into the debug block,  If the debug block is not assembled, it is not necessary in the main code.